### PR TITLE
Remote docker build support

### PR DESCRIPTION
**Pull request by Gifflen**
Due to issues with circleci 2.0 and other workflows that require docker to not to be local we can not rely on bind mounts to be able to use fleece build. This introduces fixes to allow fleece build to work with both local and remote docker servers.

### CODE REVIEW

File: fleece/cli/build/build.py [hunk 1]
```diff
@@ -104,29 +128,67 @@ def build(args):
...
+
+    with tarfile.open(fileobj=stream, mode='w') as tar:
+        tar.add(src_dir, arcname=os.path.sep)
+    stream.seek(0)
+    src.put_archive(data=stream, path='/src')
+
+    stream_1 = BytesIO()
+
+    with tarfile.open('req.tar', mode='w') as tar:
+        tar.add(requirements_path, arcname='requirements.txt')
```
Comment (miguelgrinberg) [inline fleece/cli/build/build.py]: Does this req.tar have a purpose or is it left over from an earlier version of the code?

Comment (miguelgrinberg) [reply fleece/cli/build/build.py]: Actually nm, wasn't looking at the last commit.

Comment (Gifflen) [reply fleece/cli/build/build.py]: Hah no worries. That was there originally for testing purposes. I'm planning on squashing this down once everything looks good.

**Comment by miguelgrinberg** [comment]
LGTM. Kinda hard to review in detail how you build the volumes, but if you tested it and everything works, then this seems like a good solution.

**Pull request merged**